OpenLiberty/open-liberty
# Telemetry: More JAX-RS tests and bug fixes

New Telemetry JAX-RS tests: (all for #23448)
- Test telemetry with JAX-RS server-side async (`CompletionStage` and `@Suspended AsyncResponse`)
- Test telemetry with all HTTP methods
- Test telemetry with a selection of different HTTP response codes

Bug fixes:
- Close the scope correctly for `SERVER` spans when JAX-RS server-side async is used (product bug) (fixes #24174)
- Wrap rest async client tasks correctly when `concurrent-3.0` is enabled (product bug in restfulWS) (fixes #24235)
- Don't rely on timestamps to order spans in `InMemorySpanExporter` (test bug) (fixes #24194)

Other improvements:
- Add `SpanDataMatcher` (based on existing `SpanMatcher` for Jaeger spans) to create better error messages when spans are not as expected
- Add `TestSpans` utility to create and close test spans to reduce test boilerplate.
